tests/test_benchmarks_web_urldispatcher.py

```diff
@@ -0,0 +1,133 @@
+        False,
+        False,
+        URL(path),
+    )
+
+    return web.Request(
+        message, mock.Mock(), mock.Mock(), mock.Mock(), mock.Mock(), mock.Mock()
+    )
+
+
+def test_resolve_root_route(
+    loop: asyncio.AbstractEventLoop,
+    benchmark: BenchmarkFixture,
+) -> None:
+    """Resolve top level PlainResources route 100 times."""
+    resolve_count = 100
+
+    async def handler(request: web.Request) -> NoReturn:
+        assert False
+
+    app = web.Application()
+    app.router.add_route("GET", "/", handler)
+    request = _mock_request(method="GET", path="/")
+
+    async def run_url_dispatcher_benchmark() -> None:
+        for _ in range(resolve_count):
+            await app._router.resolve(request)
+
+    @benchmark
+    def _run() -> None:
+        loop.run_until_complete(run_url_dispatcher_benchmark())
+
+
+def test_resolve_single_fixed_url_with_many_routes(
+    loop: asyncio.AbstractEventLoop,
+    benchmark: BenchmarkFixture,
+) -> None:
+    """Resolve PlainResources route 100 times."""
+    resolve_count = 100
+
+    async def handler(request: web.Request) -> NoReturn:
+        assert False
+
+    app = web.Application()
+    for count in range(250):
+        app.router.add_route("GET", f"/api/server/dispatch/{count}/update", handler)
+    request = _mock_request(method="GET", path="/api/server/dispatch/1/update")
+
+    async def run_url_dispatcher_benchmark() -> None:
+        for _ in range(resolve_count):
+            await app._router.resolve(request)
+
+    @benchmark
+    def _run() -> None:
+        loop.run_until_complete(run_url_dispatcher_benchmark())
+
+
+def test_resolve_multiple_fixed_url_with_many_routes(
+    loop: asyncio.AbstractEventLoop,
+    benchmark: BenchmarkFixture,
+) -> None:
+    """Resolve 250 different PlainResources routes."""
+
+    async def handler(request: web.Request) -> NoReturn:
+        assert False
+
+    app = web.Application()
+    for count in range(250):
+        app.router.add_route("GET", f"/api/server/dispatch/{count}/update", handler)
+
+    requests = [
+        _mock_request(method="GET", path=f"/api/server/dispatch/{count}/update")
+        for count in range(250)
+    ]
+
+    async def run_url_dispatcher_benchmark() -> None:
+        for request in requests:
+            await app._router.resolve(request)
+
+    @benchmark
+    def _run() -> None:
+        loop.run_until_complete(run_url_dispatcher_benchmark())
+
+
+def test_resolve_dynamic_resource_url_with_many_routes(
+    loop: asyncio.AbstractEventLoop,
+    benchmark: BenchmarkFixture,
+) -> None:
+    """Resolve different a DynamicResource when there are 250 PlainResources registered."""
+
+    async def handler(request: web.Request) -> NoReturn:
+        assert False
+
+    app = web.Application()
+    for count in range(250):
+        app.router.add_route("GET", f"/api/server/other/{count}/update", handler)
+    app.router.add_route("GET", "/api/server/dispatch/{customer}/update", handler)
```

Comment:
Probably need something that generate 1000+ routes and some longer ones since that better reflects downstream use cases

Reply:
Sorry, I don't follow.

Reply:
Sorry, that was a note to self so I don't forget where I left off when I context switch back